lib/rules/disallow-newline-before-block-statements.js

```diff
@@ -199,6 +203,11 @@ module.exports.prototype = {
     }
 };
 
+function isBareBlock(node) {
+    return node.parentNode &&
+    node.parentNode.type === 'BlockStatement' || node.parentNode.type === 'Program';
```

Comment:
Ok that actually broke a lot of tests - we need to also add something like ` && Array.isArray(parentNode.body)` since even `FunctionDeclaration` has a set `body` property that is a `BlockStatement` but it's an object as compared to the `IfStatement` `body` which is an array that can contain multiple statements.